module/zfs/zfs_vnops.c

```diff
@@ -278,6 +278,10 @@ zfs_holey_common(struct inode *ip, int cmd, loff_t *off)
 	if (error == ESRCH)
 		return (SET_ERROR(ENXIO));
 
+	/* file was dirty, so fall back to using file_sz logic */
+	if (error == EBUSY)
```

Comment:
Rather than suppress this error I think it would be better to return immediately and allow it to percolate up to `zpl_llseek()` and then handle it there.